### PR TITLE
Hometask6: Redux Router

### PR DESCRIPTION
**Pull request by ketanthakkar**
Added redux router.
Added page not found on invalid URL
Modified component structure to display properly.
Created new URL on movie selection and search movie.

### CODE REVIEW

File: src/components/Movie.js [hunk 1]
```diff
@@ -1,30 +1,40 @@
...
-                </div>    
-            </section>
-            <Content movies={ movie } />
-            <Footer />
-        </article>
-    )
+class Movie extends Component {
+
+    render() {
+        const movie = this.props.movies.movieData.find(movieData => movieData.id === parseInt(this.props.match.params.id) )
```
Comment (MichalWierzba) [inline src/components/Movie.js]: I'd suggest moving this to mapStateToProps

File: src/components/MovieItem.js [hunk 2]
```diff
@@ -1,11 +1,14 @@
 import React from 'react';
 import PropTypes from 'prop-types';
+import { Link } from "react-router-dom";
 
 const MovieItem = ({movieItem}) => {
     return (
         <section className="movieitem-container">
             <figure>
-                <img className="movieitem-img" src={movieItem.poster_path} alt={movieItem.tagline}/>
+                <Link to={'/film/' + movieItem.id}>
```
Comment (MichalWierzba) [inline src/components/MovieItem.js]: You can use template string here

File: src/components/Search.js [hunk 3]
```diff
@@ -14,20 +15,22 @@ export class Search extends Component {
...
+                <input type="text" placeholder="Enter title/genre" className="white-text search-title"
                         onChange={this.handleChange}></input>
                 <div className="search-container">
                     <div className="search-selection">
                         <label className="white-text">SEARCH BY</label>
                         <button id="title-btn" className={activeSearchBy === 'title' ? 'white-text' : 'white-text nonselected-color'} onClick={this.handleSearchClick}>TITLE</button>
                         <button id="genre-btn" className={activeSearchBy === 'genres' ? 'white-text' : 'white-text nonselected-color'} onClick={this.handleSearchClick}>GENRE</button>
                     </div>
-                    <button className="white-text" onClick={this.props.fetchMovies}>SEARCH</button>
+                    <Link to={'/search/Search ' + searchStr}>
```
Comment (MichalWierzba) [inline src/components/Search.js]: `/search/${encodeURIComponent(searchStr)}`
Let's discuss it during the meeting.

File: src/components/Content.js [hunk 4]
```diff
@@ -1,21 +1,40 @@
...
-            }
-            </main>
-        </React.Fragment>
-    )
+    render() {
+        const { movies, nosearch } = this.props;
+        
+        return (
+            <React.Fragment>
+                {nosearch !== "nosearch" && <Header /> }
```
Comment (MichalWierzba) [inline src/components/Content.js]: `nosearch` is always equal to "nosearch" in your app. Maybe it can be removed?

**Review by MichalWierzba — COMMENTED** [review]
Please check if unit tests are ok.

I don't see handling the situation when user opens `/film/id` directly. Please let me know if you plan to implement it, thanks :)

File: src/components/Content.js (hunk 4 above)
Comment (ketanthakkar) [reply src/components/Content.js]: Inside movie.js file i passed nosearch to hide search section in movie detail page.

**Comment by ketanthakkar** [comment]
@thetreew Done changes as per your comment. Implemented API on movie URL and search URL. 
Filter by genres on movie detail page. 
Noted todo which need to be done in later task.

File: src/actions/index.js [hunk 5]
```diff
@@ -32,24 +42,51 @@ export const sortMovies = (sortBy) => ({
...
+        const genres = result.genres[0];
+        return genres;
+    }).then((genres) => fetch(getUrl(genres, "genres", "release_date"))
+      .then(res => res.json())
+      .then(result => {
+            dispatch(receiveMovies(result))
+      }))
+}
+
+export const fetchSearchedMovies = (pharse) => (dispatch, getState) => {
```
Comment (MichalWierzba) [inline src/actions/index.js]: `pharse` -> `phrase`

File: test/MovieItem.test.js [hunk 6]
```diff
@@ -1,6 +1,7 @@
 import React from 'react';
 import { mount } from 'enzyme';
 import MovieItem from "../src/components/MovieItem";
+import { BrowserRouter as Router } from 'react-router-dom';
```
Comment (MichalWierzba) [inline test/MovieItem.test.js]: It is recommended to use `StaticRouter` or `MemoryRouter` in tests. See https://reacttraining.com/react-router/web/guides/testing/context for details.

**Review by MichalWierzba — APPROVED** [review]
Please fix last minor comments and it's 👍

**Comment by MichalWierzba** [comment]
I found one bug. Steps to reproduce:
1. open `http://localhost:4000/film/351286`
2. click any movie from the list (see that URL is updated)

Result:
Clicked movie is not loaded.

Expected result:
Clicked movie should be loaded and displayed.

**Comment by ketanthakkar** [comment]
@thetreew Bug solved. :)

**Review by MichalWierzba — APPROVED** [review]

**Pull request merged**